src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -1344,6 +1348,15 @@ def remove_temp_test_dir(self):
             errors.append("Error removing temporary test files on {}".format(result.failed_hosts))
         return errors
 
+    def __dump_engines_stacks(self, message):
+        """Dump the engines ULT stacks if self.dump_engine_ult_on_failure is True.
+
+        Args:
+            message (str): see dump_engines_stacks()
+        """
+        if self.dump_engine_ult_on_failure:
+            self.dump_engines_stacks(message)
+
```

Comment:
Since no test calls `self.dump_engines_stacks` directly, and since the core function is in `general_utils.dump_engines_stacks` anyway, maybe we don't need both `self.__dump_engines_stacks` and `self.dump_engines_stacks` and can just handle the toggle in the existing function